snackbar_presenter/lib/snackbar_presenter.dart

```diff
@@ -11,32 +11,30 @@ class SnackBarPresenter {
   const SnackBarPresenter._();
 
   /// Present an information [SnackBar] in the provided [Scaffold].
-  static void presentInformation(GlobalKey<ScaffoldState> scaffoldKey, String message,
-      {Color iconColor}) {
+  static void presentInformation(ScaffoldState scaffoldState, String message, {Color iconColor}) {
```

Comment:
[Q] This is fine, but it needs changing all of the method calls to pass in a state instead of a key, can we have methods for each? 
Unfortunately, dart doesn't support function overloading, so they'd have to have different names
https://stackoverflow.com/questions/49734608/function-overloading-in-dart#:~:text=Function%20overloading%20is%20not%20supported%20in%20Dart%20at%20all.&text=Polymorphism%20and%20static%20methods%3A,to%20be%20used%20as%20prefix.

Reply:
Yes, it does mean changing the method calls. I found this acceptable as the key contains the state, so the change required for library users is only minor. Fortunately, the only implementation of this library (at the moment) is in ScholarPack.

Ultimately,  I feel the right way to do it is to use the `ScaffoldState`, as that is what is primarily needed to display a `SnackBar`. How that state is obtained is up to the user of the library. The key might not always be available, but the state will always be a lot easier to generate (using `Scaffold.of(context)`) than having to create a `GlobalKey` and potentially pass it down to various widgets.

Reply:
Yeah that's a good point. It's not worth having two different calls. 
Merge away!